### ISSUE TITLE
Error and Lost ZCL

### ISSUE THREAD

**Issue by jamiejackson** [z-classic/electrum-zcl#149]
TLDR: I got some errors during transfers out, stuff got weird, and now I'm missing funds and my wallet balance seems to be out of sync with the wallet addresses.

Details:

Just after the snapshot today, I was trying to get ~49.99 ZCL (which was the balance in that address) from an electrum wallet (1.0.3) address to exchanges.

(For what it's worth, I also had ~5 ZCL in other addresses.)

I was working quickly, but here's what I think happened:

**First transaction**

I attempted to send ZCL to exchange A. In order to get priority, I manually entered a large fee, something like `5000`. I didn't copy it down exactly because I was in a hurry, but I got an error like `error: The transaction was rejected by network rules` followed by some JSON.

I was working quickly, but as far as I remember, my address still showed ~49.99 afterwards. _No entry for this transaction showed up in the history._

**Second transaction**

I attempted a repeat of the first, and got the same error. I confirmed that neither transaction showed up in the history.

**Third transaction**

I sent 20 ZCL to exchange B, and figuring that my large, manually-entered fee had been the problem on the last two attempts, I just used the slider and maxed out the fee that way. This transaction seemed like it went through normally, and its entry in the history shows:

Date | Amount | Balance
----- | ------- | ------
`2018-02-28 18:24` | `-20,00112298` | `34,64653297`

Full disclosure, in case it matters: I botched one of these transactions and accidentally sent it to a zcash address (yes, zcash and not zclassic). My mistake, and I can say goodbye to it

I aside from private keys, I'm not exactly sure what's sensitive and what's not in a wallet, so I've blurred out some things, but the items in my address screen don't reflect the balance that the history and (bottom of the) addresses tab are showing.

![image](https://user-images.githubusercontent.com/479475/36824803-3f177ce0-1cd2-11e8-87bd-ba8e74689949.png)

And when I access the transaction details, I see:

![image](https://user-images.githubusercontent.com/479475/36825036-63084cd2-1cd3-11e8-9b2c-264dea8a08a8.png)

I think that's trying to tell me that there are two transactions in there? That doesn't make sense.

FYI, the 20 ZCL listed at the top is unfortunately that botched (zcash) address. However, the ~29 ZCL line that's shown there is some address I never entered, and I'm not sure what that line represents.

**Comment by jamiejackson** [comment]
Hey good news, I think. I just discovered the "change" filter, and I see ~29 ZCL in there. Now I need to read https://bitcoin.stackexchange.com/a/50055 a few more times to understand it.

Looking at the bright side after having thought 50 ZCL vaporized (and only 20 really did). I'm pretty relieved.

**Comment by EmbeddedAndroid** [comment]
Curious reading this thread, could use import zcash private keys in a zclassic wallet, and recover your lost ZCL? Just a thought!

**Comment by ch4ot1c** [comment]
@jamiejackson glad to hear you're getting familiar with change addresses. Is your balance all accounted for then?

@EmbeddedAndroid please use only zclassic addresses+keys with this wallet

**Comment by jamiejackson** [comment]
@EmbeddedAndroid thanks for trying to think of a solution, but there are a couple of problems with it:
* AFAIK, if you send to an address of the wrong currency, it's just gone. 
* Even if that weren't the case, I don't have the private key to that (Cryptopia) address.

It would be nice if addresses had currency identifiers built-in (and if wallets verified it before transacting), but maybe there's some technical reason why they can't.

@ch4ot1c, yeah, I'm all set, thanks.
